docs/tutorials/gluon/mnist.md

```diff
@@ -1,333 +1,434 @@
 
-We will initialize the network parameters as follows:
+The following code defines a CNN architecture called *LeNet*.
+The LeNet architecture is a popular network known to work well on digit classification tasks.
+We will use a version that differs slightly from the original in the usage of `tanh` activations instead of `sigmoid`.
 
 ```python
-# set the context on GPU is available otherwise CPU
-ctx = [mx.gpu() if mx.test_utils.list_gpus() else mx.cpu()]
-net.initialize(mx.init.Xavier(magnitude=2.24), ctx=ctx)
-trainer = gluon.Trainer(net.collect_params(), 'sgd', {'learning_rate': 0.03})
+lenet = nn.Sequential('LeNet')
+lenet.add(
+    nn.Conv2D(channels=20, kernel_size=(5, 5), activation='tanh'),
+    nn.MaxPool2D(pool_size=(2, 2), strides=(2, 2)),
+    nn.Conv2D(channels=50, kernel_size=(5, 5), activation='tanh'),
+    nn.MaxPool2D(pool_size=(2, 2), strides=(2, 2)),
+    nn.Flatten(),
+    nn.Dense(500, activation='tanh'),
+    nn.Dense(10, activation='tanh'),
+)
 ```
 
-#### Training
+To get an overview of all intermediate sizes of arrays and the number of parameters in each layer, the `summary()` method can be a great help.
+It requires the network parameters to be initialized, and an input array to infer the sizes.
 
 ```python
-# Use Accuracy as the evaluation metric.
-metric = mx.metric.Accuracy()
-softmax_cross_entropy_loss = gluon.loss.SoftmaxCrossEntropyLoss()
-
-for i in range(epoch):
-    # Reset the train data iterator.
-    train_data.reset()
-    # Loop over the train data iterator.
-    for batch in train_data:
-        # Splits train data into multiple slices along batch_axis
-        # and copy each slice into a context.
-        data = gluon.utils.split_and_load(batch.data[0], ctx_list=ctx, batch_axis=0)
-        # Splits train labels into multiple slices along batch_axis
-        # and copy each slice into a context.
-        label = gluon.utils.split_and_load(batch.label[0], ctx_list=ctx, batch_axis=0)
-        outputs = []
-        # Inside training scope
-        with ag.record():
-            for x, y in zip(data, label):
-                z = net(x)
-                # Computes softmax cross entropy loss.
-                loss = softmax_cross_entropy_loss(z, y)
-                # Backpropogate the error for one iteration.
-                loss.backward()
-                outputs.append(z)
-        # Updates internal evaluation
-        metric.update(label, outputs)
-        # Make one step of parameter update. Trainer needs to know the
-        # batch size of data to normalize the gradient by 1/batch_size.
-        trainer.step(batch.data[0].shape[0])
-    # Gets the evaluation result.
-    name, acc = metric.get()
-    # Reset evaluation result to initial state.
-    metric.reset()
-    print('training acc at epoch %d: %s=%f'%(i, name, acc))
+lenet.initialize(mx.init.Xavier(), ctx=ctx)
+lenet.summary(nd.zeros((1, 1, 28, 28), ctx=ctx))
+# Output:
```

Comment:
I guess here the simplest is to simply put all the summary in a
` ``` ` block, rather than commented lines. Unfortunately `<!--notebook-skip-line-->` only works for single line rather than blocks 😭 so there are no ideal solution